### PR TITLE
Added Files 

### PR DESCRIPTION
**Pull request by galz10**

### CODE REVIEW

**Comment by galz10** [comment]
@gguuss

**Comment by galz10** [comment]
@hongalex

File: functions/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const app = express();
+
+var bool = false;
+
+const iotClient = new iot.v1.DeviceManagerClient({
+    // optional auth parameters.
+  });
+
+
+async function checkDevice(deviceID,pkey, res){
```
Comment (gguuss) [inline functions/index.js:18]: inconsistent spacing between parameters, recommend space between closing paren and open brace:

```node
async function checkDevice(deviceId, pKey, res) {
```

Comment (gguuss) [reply functions/index.js:18]: You probably should use a lint tool here, I believe the Node team runs [prettier](https://prettier.io/), e.g. 

```
npm install prettier
prettier --write index.js
```

File: functions/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+app.use('/added',(request,response) => {
+    console.log(" ");
+    console.log(request.body["deviceID"]);
+    console.log(request.body["publicKey"]);
+    creatDevice(request.body["deviceID"],request.body["publicKey"],response);
+    console.log(" ");
+    console.log(bool);
+    if(bool === false){
+        console.log(bool);
+
```
Comment (gguuss) [inline functions/index.js:81]: extra newline

File: functions/index.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        const response = responses[0];
+        console.log('Created device', response);
+        res.redirect('/added.html');
+        } catch (err) {
+        console.error('Could not create device', err);
+        }
+      }
+
+}
+
```
Comment (gguuss) [inline functions/index.js:58]: nit, I generally prefer two newlines between functions

File: functions/index.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+async function creatDevice(deviceID,pkey,response) {
```
Comment (gguuss) [inline functions/index.js:59]: spelling, `createDevice`

File: functions/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    var r = await(checkDevice(deviceID,pkey,response));
+
```
Comment (gguuss) [inline functions/index.js:61]: extra newline?

File: functions/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
+const functions = require('firebase-functions');
```
Comment (gguuss) [inline functions/index.js:1]: Add apache header

File: public/404.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
```
Comment (gguuss) [inline public/404.html:1]: add Apache header

File: public/added.html [hunk 8]
```diff
@@ -0,0 +1,99 @@
+<!DOCTYPE html>
```
Comment (gguuss) [inline public/added.html:1]: add header

File: style/styles.css [hunk 9]
```diff
@@ -0,0 +1,91 @@
+html {
```
Comment (gguuss) [inline style/styles.css:1]: add license header

File: readme [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (gguuss) [inline readme:1]: add instructions for deploying / testing / running

File: functions/ec_private.pem [hunk 11]
```diff
@@ -0,0 +1,5 @@
+-----BEGIN EC PRIVATE KEY-----
```
Comment (hongalex) [inline functions/ec_private.pem:1]: This file should not be committed. I would recommend deleting this and adding *.pem to your gitignore.

File: functions/package-lock.json [hunk 12]
```diff
@@ -0,0 +1,2094 @@
+{
```
Comment (gguuss) [inline functions/package-lock.json:1]: I'm not sure this file needs to be checked in

File: style/styles.css [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+  margin: 4px 2px;
+  transition-duration: 0.4s;
+  cursor: pointer;
+}
+
+.button:hover {
+  background-color: rgba(66,133,244);
+  color: white;
+}
+
```
Comment (gguuss) [inline style/styles.css:88]: extra newlines

File: functions/.gitignore [hunk 14]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (hongalex) [inline functions/.gitignore:1]: Add key files to ignore here

File: functions/index.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+                publicKey: {
+                    format: 'ES256_PEM',
+                    key: Buffer.from(pkey).toString(),
+                },
+            },
+        ],
+        };
+        const request = {
+        parent: regPath,
+        device,
```
Comment (hongalex) [inline functions/index.js:44]: Some formatting issues here. Try setting up eslint + prettier to fix this.

File: readme (hunk 10 above)
Comment (hongalex) [inline readme:1]: Generally, the title of this should be `README.md`.

File: style/styles.css [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
```
Comment (hongalex) [inline style/styles.css:91]: nit: remove the extra whitespace here

File: functions/index.js (hunk 1 above)
Comment (gguuss) [inline functions/index.js:18]: Although this is for a community article, consider adding a test, [see example here](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/functions/firebase/test
).

There's an [example from IoT's NodeJS sample](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/iot/manager/system-test/manager.test.js#L126-L143) that may also be of help.

If it's taking more than 1/2 day to add, lmk.

File: functions/ec_private.pem (hunk 11 above)
[the remark this answers is not in the document]
Comment (gguuss) [reply functions/ec_private.pem:1]: This appears to be unused, you should probably delete.

Comment (galz10) [reply functions/ec_private.pem:1]: I removed these files since they aren't being used

**Pull request merged**